### ISSUE TITLE
Ability to update the test run references

### ISSUE THREAD

**Issue by RiceChrisPeace** [gurock/trcli#170]
### What would you like the TestRail CLI to be able to do?

I want to be able to update the reference of the test run when uploading test results from CI.  

### Why is this feature necessary on the TestRail CLI?

Seeing that the test case references can be updated by uploading the test report, I find it kind of strange why we can't update the test run's reference. Because linking a test run to a ticket is then automated and I do not have to manually edit the test run & add the reference in order to view the test run results on a Jira ticket. For developers and project managers who do not have access to TestRail, this is a game changer.

### More details

Adding it to the script would be perfect:
```
trcli -y \
-h "https://testrail.net" \
--project "Integration Project" \
parse_junit -f "e2e/results.xml" \
--title "Playwright Automated Test Run ($BRANCH_NAME)" \
--case-matcher "name" \
--test-run-ref "$TICKET_REF"
```

### Interested in implementing it yourself?

Maybe, let's talk!

**Comment by bitcoder** [comment]
@RiceChrisPeace , thanks for your suggestion. To understand a bit more of your use case, what issue type do you aim to link your test run to in Jira? Is it a Task, a Story, other?
Are all those tests connected to a certain feature?

**Comment by RiceChrisPeace** [comment]
> @RiceChrisPeace , thanks for your suggestion. To understand a bit more of your use case, what issue type do you aim to link your test run to in Jira? Is it a Task, a Story, other? Are all those tests connected to a certain feature?

@bitcoder 
1. So in our process we run the tests against each branch. Each branch equals a ticket. So the tickets could be any type. The most common types would be bug/story.

2. The tests are regression tests, so all of them run against each branch. So it is not necessarily connected to a specific feature. (Feature-specific tests can be run for each ticket if we want to but that is not part of the current process)

So in summary, I want to display all the regression test outcomes on a ticket, this way it is far easier and quicker to monitor test results that will affect the next action(s) for that specific ticket.

**Comment by psd-invitaintelligence** [comment]
Has there been any update on this as I have the same issue where I want to be able to set the Jira reference in the Reference field of a test run each time the automated tests are executed?

**Comment by ecollins666** [comment]
Hi @RiceChrisPeace 
Thank you for submitting your idea for improving the CLI at TestRail! We’ve added it to the backlog for prioritisation and will review it as part of our roadmap planning.
